python_modules/dagster/dagster_tests/components_tests/executable_component_tests/test_asset_check_only.py

```diff
@@ -0,0 +1,70 @@
+from dagster._core.definitions.asset_check_result import AssetCheckResult
+from dagster._core.definitions.materialize import materialize
+from dagster._core.definitions.result import MaterializeResult
+from dagster.components.core.context import ComponentLoadContext
+from dagster.components.lib.executable_component.component import ExecutableComponent
+
+
+def only_asset_check_execute_fn(context):
+    return AssetCheckResult(passed=True)
+
+
+def test_parse_asset_check_attributes() -> None:
+    component = ExecutableComponent.from_attributes_dict(
+        attributes={
+            "name": "op_name",
+            "execute_fn": "dagster_tests.components_tests.executable_component_tests.test_asset_check_only.only_asset_check_execute_fn",
+            "checks": [
+                {
+                    "asset": "asset",
+                    "name": "check_name",
+                }
+            ],
+        }
+    )
+    assert len(component.checks or []) == 1
+    assert component.checks
+    assert component.checks[0].name == "check_name"
+    assert component.checks[0].asset == "asset"
+
+    assert component.execute_fn(None).passed is True
+
+
+def asset_and_check_execute_fn(context):
+    return MaterializeResult(
+        check_results=[AssetCheckResult(passed=True, check_name="check_name")],
+    )
+
+
+def test_execute_asset_with_check() -> None:
+    component = ExecutableComponent.from_attributes_dict(
+        attributes={
+            "name": "op_name",
+            "execute_fn": "dagster_tests.components_tests.executable_component_tests.test_asset_check_only.asset_and_check_execute_fn",
```

Comment:
could this be `.asset_and_check_execute_fn` or does that not quite work? i guess that should never happen in a real yaml file - unless its an  `__init__.py` next to it?